vic/vic_run/src/compute_irrig_demand.c

```diff
@@ -53,15 +53,17 @@ compute_irrig_demand(cell_data_struct  *cell,
             irr_sm_thresh = soil_con->max_moist[thresh_idx];
         else if (veg_lib->ithresh == IRR_FC)
             irr_sm_thresh = soil_con->Wcr[thresh_idx] / 0.7;
-        else
+        else if (veg_lib->ithresh == IRR_WP)
+            irr_sm_thresh = soil_con->Wpwp[thresh_idx]; // wilting point in urban area
             irr_sm_thresh = soil_con->Wcr[thresh_idx]; // critical point in most cases
```

Comment:
It looks like you changed the logic here incorrectly in a few ways:
1. you added your `Wpwp` case with an `else if` (which is ok) but removed the `else` of another case (the `Wcr` case) - this makes it look like the `Wcr` line is part of your new `Wpwp` case. But it isn't supposed to be.
2. even though we don't want the `Wpwp` and `Wcr` cases to be combined, let's pretend that we did want to combine them - then you would have 2 lines following the `else if (...)` line.  But in C, if your case has more than 1 line, you MUST enclose it in curly brackets `{ }`. So this would still be a mistake. As it is, without the `{ }`, the `Wcr` line will be interpreted as not inside ANY `if ... else` blocks. This means it will get executed all the time. This means that `irr_sm_thresh` will get set to the `Wcr` value after you set it to the `Wpwp` value. It will ALWAYS end up being set to the `Wcr` value.

So, what you need to have is:
line    code
56     `else if (veg_lib->ithresh == IRR_CR)`  (this is REPLACING the `else` that originally was there
57     `    irr_sm_thresh = soil_con->Wcr[thresh_idx];`
58     `else`  or you could have `else if (veg_lib->ithresh == IRR_WP)`, but I think it is good to end with `else` so that we always set `irr_sm_thresh` to something
59     `    irr_sm_thresh = soil_con->Wpwp[thresh_idx];`